core/templates/pages/exploration-editor-page/modal-templates/lost-changes-modal.controller.spec.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for LostChangesModalController.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { LostChangeObjectFactory } from
+  'domain/exploration/LostChangeObjectFactory';
+import { TestBed } from '@angular/core/testing';
+import { UtilsService } from 'services/utils.service';
+// ^^^ This block is to be removed.
+
+describe('Lost Changes Modal Controller', () => {
+  let $scope = null;
+  let $uibModalInstance = null;
+  let $log = null;
+  let logSpy = null;
+  const explorationId = '0';
+  const lostChanges = [{
+    cmd: 'add_state',
+    state_name: 'State name',
+  }];
+
+  beforeEach(function() {
+    TestBed.configureTestingModule({
+      providers: [
+        LostChangeObjectFactory
+      ]
+    });
+  });
+
+  beforeEach(angular.mock.module('oppia', ($provide) => {
+    $provide.value('LostChangeObjectFactory', (
+      new LostChangeObjectFactory(new UtilsService)));
```

Comment:
@marianazangrossi Instead of doing this, you could add `UtilsService` to providers, and use `TestBed.get(LostChangeObjectFactory)`. Here it is one dependency, but if the dependencies increase (both horizontally and vertically) it would become really difficult and it will soon resemble upgraded services, the very thing we are trying to remove😂. Maybe you could just add the services you need to TestBed. You can either just provide them as is or incase there are too many dependencies mock them with `spies` or mock classes and just provide the value here. What do you think?

Reply:
Sure, I think it makes sense. I missed this part! I've fixed now.

Reply:
But I'm not sure if I got it correctly. If you think it's not right yet, please comment out!

Reply:
Nope, this is perfect. Thank you!